interfaces/builtin/usb_gadget.go

```diff
@@ -29,25 +48,199 @@ const usbGadgetBaseDeclarationSlots = `
     deny-auto-connection: true
 `
 
-const usbGadgetConnectedPlugAppArmor = `
-# https://www.kernel.org/doc/Documentation/usb/gadget_configfs.txt
-# Allow creating new gadgets under usb_gadget, which is creating
-# new directories
-/sys/kernel/config/usb_gadget/ rw,
-# Allow creating sub-directories, symlinks and files under those
-# directories
-/sys/kernel/config/usb_gadget/** rw,
-
-# Allow access to UDC
-/sys/class/udc/ r,
+const usbGadgetConnectedPlugSecComp = `
+# Description: Allow mount and umount syscall access. No filtering here, as we
+# rely on AppArmor to filter the mount operations.
+mount
+umount
+umount2
 `
 
+// usbGadgetInterface allows creating transient and persistent mounts
+type usbGadgetInterface struct {
+	commonInterface
+}
+
+type ffsMountInfo struct {
+	name       string
+	where      string
+	persistent bool
+}
+
+func (mi *ffsMountInfo) validate() error {
+	// for ffs the name is the name of the function, which is not a path so we
+	// just need to ensure it doesn't contain any AppArmor regex characters.
+	if err := validateNoAppArmorRegexpWithError(`cannot use usb-gadget "name" attribute`, mi.name); err != nil {
+		return err
+	}
+
+	// Reuse the where validation from mount-control, as the semantics are the same
+	if err := validateWhereAttr(mi.where); err != nil {
+		return err
+	}
+
+	// Until we have a clear picture of how this should work, disallow creating
+	// persistent mounts into $SNAP_DATA
+	if mi.persistent && strings.HasPrefix(mi.where, "$SNAP_DATA") {
+		return errors.New(`mount-control "persistent" attribute cannot be used to mount onto $SNAP_DATA`)
+	}
+
+	return nil
+}
+
+func ffsMounts(plug interfaces.Attrer) ([]map[string]any, error) {
+	var mounts []map[string]any
```

Comment:
I was expecting a concrete type here. Why do we need to use `any`?

Reply:
It was suggested I use any instead `interface{}` --- or what is your question exactly?